test/socket.jl

```diff
@@ -255,10 +255,14 @@ let
     end
     a, b, c = [create_socket() for i = 1:3]
     try
-        @sync begin
-            send(c, bcastdst, 2000, "hello")
-            recvs = [@async @test String(recv(s)) == "hello" for s in (a, b)]
-            map(wait, recvs)
+        # bsd family do not allow broadcasting to ip"255.255.255.255"
+        # or ip"127.255.255.255"
+        @static if !is_bsd() || is_apple()
+            @sync begin
+                send(c, bcastdst, 2000, "hello")
+                recvs = [@async @test String(recv(s)) == "hello" for s in (a, b)]
+                map(wait, recvs)
```

Comment:
@amitmurthy thanks your snippet!
I committed it @ 0d97e7f.

The timer works correctly:
```
socket: Error During Test
  Got an exception of type LoadError outside of a @test
  LoadError: timeout
  Stacktrace:
   [1] (::Test73Main_socket.#wait_until_timeout#28)(::Array{Task,1}) at /usr/home/iblis/git/julia/test/socket$
jl:265                                                                                                           
[2] include_from_node1 at ./loading.jl:539
[3] include at ./sysimg.jl:14                                                                                 
[4] macro expansion at /usr/home/iblis/git/julia/test/testdefs.jl:18 [inlined]                                [5] macro expansion at ./test.jl:853 [inlined]                                                                
[6] macro expansion at ./util.jl:378 [inlined]                                                                
[7] macro expansion at /usr/home/iblis/git/julia/test/testdefs.jl:0 [inlined]                                 [8] anonymous at ./<missing>:?                                                                                
[9] eval at ./boot.jl:235
[10] runtests(::String, ::Bool) at /usr/home/iblis/git/julia/test/testdefs.jl:21                              
...
```